src/Symfony/Component/Console/Tester/CommandTester.php

```diff
@@ -24,9 +27,31 @@ class CommandTester
 {
     use TesterTrait;
 
+    private Command $command;
+
     public function __construct(
-        private Command $command,
+        callable|Command $command,
     ) {
+        if (!$command instanceof Command) {
+            if (!\is_object($command) || $command instanceof \Closure) {
+                throw new InvalidArgumentException(\sprintf('The command must be an instance of "%s" or an invokable object.', Command::class));
+            }
+
+            /** @var AsCommand $attribute */
+            $attribute = ((new \ReflectionObject($command))->getAttributes(AsCommand::class)[0] ?? null)?->newInstance()
+                ?? throw new LogicException(\sprintf('The command must use the "%s" attribute.', AsCommand::class));
+
+            $command = (new Command($attribute->name))
+                ->setDescription($attribute->description ?? '')
+                ->setHelp($attribute->help ?? '')
+                ->setCode($command);
+
+            foreach ($attribute->usages as $usage) {
+                $command->addUsage($usage);
+            }
```

Comment:
re-checking the new `Application::addCommand` again, it looks the same code, wondering if we can create a new static factory method `Command::createFromInvokable($invokable)` to avoid code duplication?

Reply:
Cc @GromNaN

Reply:
In would not add a new "technical" method to the command class that is already very loaded. If we want to factorize this code, I think a attribute class would be appropriate:

```php
class AsCommand
{
    public static function createCommand(callable&object $code): Command
}

Reply:
I'll follow up later

Reply:
I pushed the required change but I can't get it to work, mostly because I don't know how the `InvokableCommand` wrapping works.

@yceruto would you be able to check it out?